File: frontend/src/models/taskrule.ts
export interface ITaskRule {
    guid: string;
    name: string;
    description: string;
    action: string;
    ruleConditions: any;
    whitelist?: any;
    blacklist?: any;
    isActive: boolean;
    createdAt: Date;
    updatedAt: Date;
}

export class TaskRule implements ITaskRule {
    guid: string;
    name: string;
    description: string;
    action: string;
    ruleConditions: any;
    whitelist?: any;
    blacklist?: any;
    isActive: boolean;
    createdAt: Date;
    updatedAt: Date;

    constructor(data: ITaskRule) {
        this.guid = data.guid;
        this.name = data.name;
        this.description = data.description;
        this.action = data.action;
        this.ruleConditions = data.ruleConditions;
        this.whitelist = data.whitelist;
        this.blacklist = data.blacklist;
        this.isActive = data.isActive;
        this.createdAt = new Date(data.createdAt);
        this.updatedAt = new Date(data.updatedAt);
    }
}

File: backend/app/models/taskrule.py
from ..db import db


class TaskRule(db.Model):
    __tablename__ = 'task_rules'
    guid = db.Column(db.UUID, primary_key=True, unique=True)
    name = db.Column(db.String(255), nullable=False)
    description = db.Column(db.Text, nullable=False)
    action = db.Column(db.String(255), nullable=False)
    rule_conditions = db.Column(db.JSON, nullable=False)
    whitelist = db.Column(db.JSON)
    blacklist = db.Column(db.JSON)
    is_active = db.Column(db.Boolean, nullable=False, default=True)

    created_at = db.Column(db.DateTime, nullable=False, default=db.func.current_timestamp())
    updated_at = db.Column(db.DateTime, nullable=False, default=db.func.current_timestamp(),
                           onupdate=db.func.current_timestamp())
    tasks = db.relationship('Task', backref='task_rule')

    def to_dict(self):
        return {
            "guid": self.guid,
            "name": self.name,
            "description": self.description,
            "action": self.action,
            "rule_conditions": self.rule_conditions,
            "whitelist": self.whitelist,
            "blacklist": self.blacklist,
            "is_active": self.is_active,
            "created_at": self.created_at,
            "updated_at": self.updated_at
        }

    def __repr__(self):
        return f'TaskRule {self.name}'
